internal/common/normalattackstate.go

```diff
@@ -134,3 +140,146 @@ func Get5PercentN0Delay(activeChar *character.CharWrapper) int {
 
 	return percentDelay5[activeCharKey]
 }
+
+func Get0PercentN0Delay(activeChar *character.CharWrapper) int {
+	// TODO: Collect data for this
+	return 0
+}
+
+type NAHook struct {
+	c           *character.CharWrapper
+	abilName    string
+	Core        *core.Core
+	abilKey     string
+	abilProcICD int
+	abilICDKey  string
+	abilTickSrc int
+	abilHookSrc int
+	delayFunc   func(*character.CharWrapper) int
+	summonFunc  func()
+}
+
+func NewNAHook(c *character.CharWrapper, Core *core.Core, abilName string, abilKey string, abilProcICD int, abilICDKey string, delayFunc func(*character.CharWrapper) int, summonFunc func()) *NAHook {
+	return &NAHook{
+		c:           c,
+		abilName:    abilName,
+		Core:        Core,
+		abilKey:     abilKey,
+		abilProcICD: abilProcICD,
+		abilICDKey:  abilICDKey,
+		abilTickSrc: 0,
+		abilHookSrc: 0,
+		delayFunc:   delayFunc,
+		summonFunc:  summonFunc,
+	}
+
+}
+
+func (w *NAHook) NAStateHook() {
```

Comment:
NIT: I'd name this something like `Attach` or `Enable` to make it clear that it is performing a subscription